src/write.rs

```diff
@@ -468,6 +541,23 @@ mod tests {
         assert_eq!(w.oplog.len(), 2);
     }
 
+    #[test]
+    fn take_test() {
+        let (mut w, _r) = crate::new_from_empty::<i32, _>(2);
+        w.append(CounterAddOp(1));
+        w.publish();
+        w.append(CounterAddOp(1));
+        assert_eq!(w.take(), 4);
+
+        let (mut w, _r) = crate::new_from_empty::<i32, _>(2);
+        w.append(CounterAddOp(1));
+        // w.publish();
```

Comment:
I'd like to see a test with an empty oplog too, just as a sanity check. That is, one with `.append` followed by `.publish`, followed by `.take`.